### PR TITLE
test(evmstaking/keeper): add test cases for staking queue

### PR DESCRIPTION
**Pull request by zsystm**
Increased test coverage from 0% to 84.6%
- **What's Covered**
  - Most application logics written on the Story side, excluding SDK.
- **What’s Not Covered (Edge Cases Related to the SDK)**
  - failed to get iterator
  - failed to unmarshal

issue: #64

### CODE REVIEW

File: client/x/evmstaking/keeper/staking_queue_test.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	s.setupValidatorAndDelegation(ctx, valPubKey, delPubKey, valAddr, delAddr)
+	s.setupValidatorAndDelegation(ctx, valPubKey2, delPubKey, valAddr2, delAddr)
+
+	tcs := []struct {
+		name           string
+		setUnbondings  func(c context.Context)
+		expectedResult []stypes.DVPair
+		expectedError  string
+	}{
+		{
```
Comment (limengformal) [inline client/x/evmstaking/keeper/staking_queue_test.go:90]: Can we add a test case with mixed items of items within the unbonding period and items after the unbonding period?

Comment (zsystm) [reply client/x/evmstaking/keeper/staking_queue_test.go:90]: @limengformal 

image

In the situation shown above, you’re referring to a test case where GetMaturedUnbondings returns ubd1 and ubd2, right? Since ubd3 and ubd4 have not yet matured, they should not appear in the result of GetMaturedUnbondings; only ubd1 and ubd2 should be shown explicitly

Comment (limengformal) [reply client/x/evmstaking/keeper/staking_queue_test.go:90]: Yes, that is the scenarios I was referring to. Let's add that

Comment (zsystm) [reply client/x/evmstaking/keeper/staking_queue_test.go:71]: @limengformal 
I have rebased latest main and add [mixed test case](https://github.com/piplabs/story/pull/68/files#diff-60aae4ca5374cd765a3a9b121e416c91c5bbbf2815b2907965feb46c164cd346R79-R88).

File: client/x/evmstaking/keeper/staking_queue_test.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+			name: "pass: no matured unbondings",
+			setUnbondings: func(c context.Context) {
+				s.setupUnbonding(c, delAddr, valAddr1, "100")
+				s.setupUnbonding(c, delAddr, valAddr2, "100")
+			},
+			expectedResult: nil,
+		},
+		{
+			name: "pass: one mautred and one not matured unbonding",
```
Comment (limengformal) [inline client/x/evmstaking/keeper/staking_queue_test.go:80]: [nit] mautred -> matured

**Review by limengformal — COMMENTED** [review]
@zsystm Let's rebase again

File: client/x/evmstaking/keeper/staking_queue_test.go (hunk 2 above)
Comment (zsystm) [reply client/x/evmstaking/keeper/staking_queue_test.go]: @limengformal 
I have rebased and fixed typo. Thanks

**Review by limengformal — APPROVED** [review]
